fuel-tx/src/transaction/types/input.rs

```diff
@@ -455,9 +455,9 @@ impl Input {
             Input::CoinPredicate(_) => InputRepr::Coin.coin_predicate_offset(),
             Input::MessageCoinPredicate(_) => InputRepr::Message.data_offset(),
             Input::MessageDataPredicate(MessageDataPredicate { data, .. }) => {
-                InputRepr::Message
-                    .data_offset()
-                    .map(|o| o + bytes::padded_len(data))
+                InputRepr::Message.data_offset().map(|o| {
+                    o.saturating_add(bytes::padded_len(data).unwrap_or(usize::MAX))
```

Comment:
The `usize::Max` kinda makes more sense here with the `saturating_add`, but I haven a hard time with `None` for len being equivalent to `usize::Max`.

You could do a:
```rs
bytes::padded_len(data).map(|len| InputRepr::Message.data_offset().map(|o| o.saturating_add(len))
```
Not really an improvement on readability though :/ So up to you.